### PR TITLE
doc: Proposal for Samples QA

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
Proposal for we improve and ensure the QA of Samples

### CODE REVIEW

File: doc/proposals/samples.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+- Ansible and Go Memcached samples should be integrated with [Coveralls](https://coveralls.io/)
+- All above stories should be achieved successfully
+
+### Risks and Mitigations
+
+May [Coveralls](https://coveralls.io/) do not work well with [molecule](https://github.com/operator-framework/operator-sdk-samples/tree/master/ansible/memcached-operator/molecule) and then, we can just not integrate it with or find another similar tool.
+
+## Open Questions 
+
+How to do the same for Helm operators?
```
Comment (camilamacedo86) [inline doc/proposals/samples.md]: @joelanford have we any doc over how to cover helm operators with tests?

Comment (estroz) [reply doc/proposals/samples.md]: Yes in [doc/helm/dev/developer_guide.md](https://github.com/operator-framework/operator-sdk/blob/master/doc/helm/dev/developer_guide.md#testing-a-helm-operator-locally).

Comment (camilamacedo86) [reply doc/proposals/samples.md]: Hi @estroz,

Thank you for the doc. But it is how to run the operator locally and/or in the cluster. I mean how to test the operator with tests/checks unit and integration tests. Do you know that?

Comment (joelanford) [reply doc/proposals/samples.md]: I'd start with this: https://github.com/operator-framework/operator-sdk/blob/master/hack/tests/e2e-helm.sh

File: doc/proposals/samples.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+- The tests should be working successfully
+- The project should have an small section over how to test and with links for reference
+- The project should have a makefile target to call the tests
+
+#### tests for Helm Memcached Sample
+
+- I as a Helm dev operator user, I'd like to how to cover the projects with tests
+
+**Acceptance Criteria** 
+- The Helm project should have a few tests as using shell as it done [here](https://github.com/operator-framework/operator-sdk/blob/master/hack/tests/e2e-helm.sh).
```
Comment (jmrodri) [inline doc/proposals/samples.md]: Reword:
`The Helm project should have test shell scripts as seen [here](https://github.com/operator-framework/operator-sdk/blob/master/hack/tests/e2e-helm.sh).`

File: doc/proposals/samples.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+- The tests should be working successfully
+- The project should have an small section over how to test and with links for reference
```
Comment (jmrodri) [inline doc/proposals/samples.md]: Same comment from the Go and Ansible section applies to these two.

File: doc/proposals/samples.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+- The project should have a makefile target to call the tests
+
+
+#### e2e tests for Ansible Memcached Sample
+
+- I as a Ansible dev operator user, I'd like to how to cover the projects with integration tests using molecule
```
Comment (jmrodri) [inline doc/proposals/samples.md]: Change to:
`As an Ansible operator developer, I would ...`

File: doc/proposals/samples.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+**Acceptance Criteria** 
+- The Ansible project should have a few integration tests using [molecule](https://github.com/operator-framework/operator-sdk-samples/tree/master/ansible/memcached-operator/molecule) which by default is scaffold
+- The tests should be working successfully
+- The project should have an small section over how to test and with links for reference
```
Comment (jmrodri) [inline doc/proposals/samples.md]: See the sections above.

File: doc/proposals/samples.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+- The project should have a makefile target to call the tests
+
+#### Travis Integration
+
+- I as a maintainer, I'd like to ensure that the samples still working after the changes performed in the PR easily
```
Comment (jmrodri) [inline doc/proposals/samples.md]: * maintainer of what? please clarify in the statement.
* Change to: `As a maintener INSERT WHAT, I would like ...`
* rephrase `ensure that samples continue to work after the changes performed in the PR are applied.`

File: doc/proposals/samples.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+- I as a dev operator user, I'd like to know how to call the tests in the CI and integrate them with Travis
```
Comment (jmrodri) [inline doc/proposals/samples.md]: change to: 
`As an operator developer, I would ...`

File: doc/proposals/samples.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+**Acceptance Criteria** 
+- The Ansible, Helm and Go project should be integrated with Travis
+- All PR's made against the master should trigger th CI
```
Comment (jmrodri) [inline doc/proposals/samples.md]: rephrase
`All PRs against the master branch should trigger the CI`

File: doc/proposals/samples.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+**Acceptance Criteria** 
+- The Ansible, Helm and Go project should be integrated with Travis
+- All PR's made against the master should trigger th CI 
+- The unit and integration tests should be checked in the CI
+- The CI should fail if one of the tests do not pass
+
+#### Coveralls Integration
+
+- I as a dev operator user, I'd like to know good practices to ensure the quality of my operators projects
```
Comment (jmrodri) [inline doc/proposals/samples.md]: Change to:
`As an operator developer, I would ...`

File: doc/proposals/samples.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+- I as a dev operator user, I'd like to know good practices to ensure the quality of my operators projects 
+
+**Acceptance Criteria** 
+- Ansible and Go Memcached samples should be integrated with [Coveralls](https://coveralls.io/)
+- All above stories should be achieved successfully
+
+### Risks and Mitigations
+
+May [Coveralls](https://coveralls.io/) do not work well with [molecule](https://github.com/operator-framework/operator-sdk-samples/tree/master/ansible/memcached-operator/molecule) and then, we can just not integrate it with or find another similar tool.
```
Comment (jmrodri) [inline doc/proposals/samples.md]: Rephrase (keep the links though)
`Coveralls may not work well with molecule, if this is the case we can just not integrate with it or we can find a similar tool`

**Pull request merged**